lib/services/storageManagement2/lib/operations/storageAccounts.js

```diff
@@ -52,10 +86,34 @@ function StorageAccounts(client) {
  *                      {object} [request]  - The HTTP Request object if an error did not occur.
  *
  *                      {stream} [response] - The HTTP Response stream if an error did not occur.
+ * 
+ *          {Promise}   A promise is returned.
+ *             @resolve {CheckNameAvailabilityResult} - The deserialized result object.
+ *             @reject {Error} - The error object.
  */
-StorageAccounts.prototype.checkNameAvailability = function (name, options, callback) {
+StorageAccounts.prototype.checkNameAvailability = function (name, options, optionalCallback) {
```

Comment:
The method signature remains the same. callback is now optional. If the customer does not provide a callback then we return a Promise.